File: projects/order-creation-queries/repository/src/main/java/com/robinfood/repository/paymentmethods/IPaymentMethodsRepository.java
package com.robinfood.repository.paymentmethods;

import com.robinfood.core.dtos.PaymentMethodsFilterDTO;
import com.robinfood.core.enums.Result;

import java.util.List;

/**
 * Repository Payment methods
 */
public interface IPaymentMethodsRepository {

    /**
     * Get object with the list of payment methods
     *
     * @param token token auth service
     * @return list of payment methods filtered by status
     */
    Result<List<PaymentMethodsFilterDTO>> getDataPaymentMethods(
            String token
    );

}

File: projects/posv2/repository/src/main/kotlin/com/robinfood/repository/paymentmethods/IPaymentMethodsRepository.kt
package com.robinfood.repository.paymentmethods

import com.robinfood.core.dtos.paymentmethods.PaymentMethodResponseDTO
import com.robinfood.core.enums.Result

/**
 * Repository for payment methods related data
 */
interface IPaymentMethodsRepository {

    /**
     * Retrieves the store's payment methods
     * [channelId] channel id
     * [originId] the origin identification to get payment methods
     * [storeId] the store identification to get payment methods
     * [token] the authentication token to be used
     * @return the active delivery platforms for the store
     */
    suspend fun getPaymentMethods(
            channelId: Long,
            originId: Long,
            storeId: Long,
            token: String
    ): Result<List<PaymentMethodResponseDTO>>
}